lib/promscrape/config.go

```diff
@@ -1144,6 +1144,9 @@ func (sdc *FileSDConfig) appendScrapeWork(dst []*ScrapeWork, swsMapPrev map[stri
 				if sws := swsMapPrev[path]; sws != nil {
 					// Re-use the previous valid scrape work for this path.
 					logger.Errorf("keeping the previously loaded `static_configs` from %q because of error when re-loading the file: %s", path, err)
+					// Slice of destinations should be cleared because if we add previously got scrape work, it will contain duplicates.
```

Comment:
Please don't explain 'why do you do that' add just 'what do you do here'

Reply:
updated